### ISSUE TITLE
IRCv3 history support?

### ISSUE THREAD

**Issue by poVoq** [itskrystibitch/websocketirc#1]
Would be cool if this would allow back-scrolling older history via the new IRCv3 history specs.

KiwiIRC and Gamja support this already with an IRCd like Ergo.chat without any bouncer, strait via websockets.

https://ircv3.net/specs/extensions/chathistory

**Comment by itskrystibitch** [comment]
This widget not only doesn't support IRCv3, it doesn't even have private messages, modes, kick, ban, etc. It's intended for a TINY (18KB'ish) widget in a website widget where users may be unfamiliar with IRC. The lack of some of those features being intended, such as a lack of private messages incase there is harassment factors and admins can handle it from an actual IRC client instead of the widget. Hope that clarifies THIS particular project but I do have plenty of experience in building IRC clients and there is actually more code and different IRC clients for different web purposes on the backburner. This was just the easiest, and first one going out. Some of the bonus features in this widget is supporting the posting of images and YouTube videos directly to the chat which will render the image and post the YouTube embed making that userbase of "doesn't know IRC" have an easier time congregating among each other.

Thanks for your interest.

**Comment by hellokrysti** [comment]
> Would be cool if this would allow back-scrolling older history via the new IRCv3 history specs.
> 
> KiwiIRC and Gamja support this already with an IRCd like Ergo.chat without any bouncer, strait via websockets.
> 
> https://ircv3.net/specs/extensions/chathistory

This widget not only doesn't support IRCv3, it doesn't even have private messages, modes, kick, ban, etc. It's intended for a TINY (18KB'ish) widget in a website widget where users may be unfamiliar with IRC. The lack of some of those features being intended, such as a lack of private messages incase there is harassment factors and admins can handle it from an actual IRC client instead of the widget. Hope that clarifies THIS particular project but I do have plenty of experience in building IRC clients and there is actually more code and different IRC clients for different web purposes on the backburner. This was just the easiest, and first one going out. Some of the bonus features in this widget is supporting the posting of images and YouTube videos directly to the chat which will render the image and post the YouTube embed making that userbase of "doesn't know IRC" have an easier time congregating among each other.

Thanks for your interest.

**Comment by poVoq** [comment]
Yes I understand this, but if a user is seeing this widget with a bit of channel history that would make this so much nicer for the typical non-irc user.

I don't know how complicated it would be to add chat history (only), but I think it would really add to the usefulness of this neat little widget :)

**Comment by hellokrysti** [comment]
> I don't know how complicated it would be to add chat history (only), but I think it would really add to the usefulness of this neat little widget :)

I'll consider it in this one given it doesn't bloat it up too much or at least dump a fork of this so people can choose which level of bloat they'd like to introduce to a widget. If anything changes I'll drop a reply when it's done.

**Comment by hellokrysti** [comment]
Should be all set. No need for a separate thing, it was just a lil bit added in. It's tested on UnrealIRCd v6+ so just keep that in mind that's where I did everything and I haven't currently got any plans to support other IRCd's if this doesn't work elsewhere. Channel it's used in must be set to +H in a /mode #channel +H [messages]:[duration] so /mode #channel +H 50:1440 would allow you an entire day's buffer of 50 messages. The IRC config block must be set to true where it has the ircv3 parameter and that's it. This chathistory thing is in "draft" mode I noticed so this thing may break over time through changes in UnrealIRCd. I'll do my best to update that end of it, I'm sure I'll notice but messages are welcome if anybody notices any critical changes that break this before I do.
